src/server/game/AuctionHouse/AuctionHouseMgr.cpp

```diff
@@ -648,105 +648,128 @@ void AuctionHouseObject::BuildListOwnerItems(WorldPacket& data, Player* player,
 void AuctionHouseObject::BuildListAuctionItems(WorldPacket& data, Player* player,
     std::wstring const& wsearchedname, uint32 listfrom, uint8 levelmin, uint8 levelmax, uint8 usable,
     uint32 inventoryType, uint32 itemClass, uint32 itemSubClass, uint32 quality,
-    uint32& count, uint32& totalcount)
+    uint32& count, uint32& totalcount, bool getall)
 {
     int loc_idx = player->GetSession()->GetSessionDbLocaleIndex();
     int locdbc_idx = player->GetSession()->GetSessionDbcLocale();
 
     time_t curTime = sWorld->GetGameTime();
+    time_t throttleTime = GetAllThrottleMap.find(player->GetGUID()) != GetAllThrottleMap.end() ? GetAllThrottleMap[player->GetGUID()] : curTime;
```

Comment:
You are doing two lookups for the same key in the map. This is inefficient. Use the iterator you get from .find instead.